packages/app/src/common/subscriptions/websocket.ts

```diff
@@ -0,0 +1,290 @@
+    request: IncomingMessage;
+  }
+];
+
+export const CreateSubscriptionsServer = async (
+  options: WebsocketSubscriptionsOptions | undefined
+): CommonSubscriptionsServer => {
+  const enableOldTransport = options === 'legacy' || (typeof options === 'object' && options.subscriptionsTransport);
+
+  const enableGraphQLWS = options === true || (typeof options === 'object' && options.graphQLWS);
+
+  const enableAll = enableOldTransport && enableGraphQLWS;
+
+  const enabled: 'new' | 'both' | 'legacy' | 'none' = enableAll
+    ? 'both'
+    : enableOldTransport
+    ? 'legacy'
+    : enableGraphQLWS
+    ? 'new'
+    : 'none';
+
+  if (enabled === 'none') return null;
+
+  const optionsObj =
+    typeof options === 'object'
+      ? {
+          subscriptionsTransport: typeof options.subscriptionsTransport === 'object' ? options.subscriptionsTransport : {},
+          graphQLWS: typeof options.graphQLWS === 'object' ? options.graphQLWS : {},
+          buildContext: options.buildSubscriptionsContext,
+          wsOptions: options.wsOptions,
+        }
+      : {};
+
+  const GRAPHQL_TRANSPORT_WS_PROTOCOL = 'graphql-transport-ws';
+  const GRAPHQL_WS = 'graphql-ws';
+
+  const [ws, subscriptionsTransportWs, useGraphQLWSServer] = await Promise.all([
+    import('ws').then(v => v.default),
+    enableOldTransport ? import('subscriptions-transport-ws/dist/server.js').then(v => v.SubscriptionServer) : null,
+    enableGraphQLWS ? import('graphql-ws/lib/use/ws').then(v => v.useServer) : null,
+  ]);
+
+  const wsServer: WebSocket.Server | [graphqlWsServer: WebSocket.Server, subWsServer: WebSocket.Server] = enableAll
+    ? [
+        /**
+         * graphql-ws
+         */
+        new ws.Server({
+          ...cleanObject(optionsObj.wsOptions),
+          noServer: true,
+        }),
+        /**
+         * subscriptions-transport-ws
+         */
+        new ws.Server({
+          ...cleanObject(optionsObj.wsOptions),
+          noServer: true,
+        }),
+      ]
+    : new ws.Server({
+        ...cleanObject(optionsObj.wsOptions),
+        noServer: true,
+      });
+
+  const { buildContext } = optionsObj;
+
+  return function (getEnveloped) {
+    const { schema, execute, subscribe, contextFactory, validate } = getEnveloped();
+
+    async function getContext(contextArgs: BuildSubscriptionContextArgs) {
+      if (buildContext) return contextFactory(Object.assign({}, await buildContext(contextArgs)));
+
+      return contextFactory(contextArgs);
+    }
+
+    let wsTuple: CommonSubscriptionsServerTuple;
+
+    if (enabled === 'new') {
+      assert(!Array.isArray(wsServer));
+      assert(useGraphQLWSServer);
+
+      useGraphQLWSServer(
+        {
+          ...cleanObject(optionsObj.graphQLWS),
+          execute,
+          subscribe,
+          validate,
+          async onSubscribe({ connectionParams, extra: { request, socket } }, { payload: { operationName, query, variables } }) {
+            const { schema, parse } = getEnveloped();
```

Comment:
Discussed with @dotansimha in Slack, all functions from `getEnveloped` are safe to be used statically. You can destruct the `parse` from `getEnveloped ` above and use it from there instead.

Ultimately, only `schema` is required to be reacquired because it may change dynamically.

```suggestion
            const { schema } = getEnveloped();
```